src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-pypemicro.py

```diff
@@ -0,0 +1,75 @@
+# ------------------------------------------------------------------
+# Copyright (c) 2022 PyInstaller Development Team.
+#
+# This file is distributed under the terms of the GNU General Public
+# License (version 2.0 or later).
+#
+# The full license is available in LICENSE.GPL.txt, distributed with
+# this software.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ------------------------------------------------------------------
+
+# Hook for the pypemicro module: https://github.com/nxpmicro/pypemicro
+
+import glob
+import os
+from PyInstaller.utils.hooks import remove_prefix, get_package_paths
+from PyInstaller.compat import is_win, is_linux, is_darwin
+from PyInstaller import compat
+from PyInstaller import log as logging
+
+logger = logging.getLogger(__name__)
+
+# Patterns of dynamic library filenames that might be bundled with some installed Python packages.
+PY_DYLIB_PATTERNS = [
+    '*.dll',
+    '*.dylib',
+    '*.so',
+]
+
+
+def collect_dynamic_libs(package, destdir=None):
+    """
+    This function produces a list of (source, dest) of dynamic library files that reside in package. Its output can be
+    directly assigned to ``binaries`` in a hook script. The package parameter must be a string which names the package.
+
+    :param destdir: Relative path to ./dist/APPNAME where the libraries should be put.
+    """
+    # Accept only strings as packages.
+    if not isinstance(package, compat.string_types):
+        raise TypeError('package must be a str')
+
+    logger.debug('Collecting dynamic libraries for %s' % package)
+    pkg_base, pkg_dir = get_package_paths(package)
+    # Walk through all file in the given package, looking for dynamic libraries.
+    dylibs = []
+    for dirpath, _, __ in os.walk(pkg_dir):
+        # Try all file patterns in a given directory.
+        for pattern in PY_DYLIB_PATTERNS:
+            files = glob.glob(os.path.join(dirpath, pattern))
+            for source in files:
+                # Produce the tuple (/abs/path/to/source/mod/submod/file.pyd, mod/submod/file.pyd)
+                if destdir:
+                    # Libraries will be put in the same directory.
+                    dest = destdir
+                else:
+                    # The directory hierarchy is preserved as in the original package.
+                    dest = remove_prefix(dirpath, os.path.dirname(pkg_base) + os.sep)
+                logger.debug(' %s, %s' % (source, dest))
+                dylibs.append((source, dest))
+    return dylibs
+
```

Comment:
What's the motivation for copy/pasting this function in here instead of just using `from PyInstaller.utils.hooks import collect_dynamic_libs`?

Reply:
I forgot to add a comment. The important change is in `PY_DYLIB_PATTERNS` where `lib*.so` has been changed into `*.so`
So instead of writing new (buggy) code I copied the existing just with this change.

Reply:
Yes, I wanted to avoid importing the lib. The isolated subprocess is new to me. I'll give it a try, thanks for pointing that.

Reply:
Refactored using the isolation and calling the pypemicro for libraries.